### DISCUSSION TITLE
How to use Doc.__init__ with heads

### DISCUSSION THREAD

**Discussion by pb-jeff-oneill** [explosion/spaCy#8334] [Help: Coding & Implementations] [answered]
I'm trying to create a Doc from the constructor with specified heads like this:

> nlp = spacy.blank('en')
> d = Doc(nlp.vocab, ['The', 'cat'], heads=[1,1])

The goal is for 'cat' to be the root and also the head for 'The' like [this](https://explosion.ai/demos/displacy?text=The%20cat&model=en_core_web_sm&cpu=0&cph=0).

I've tried many combinations of values for heads, but the result of 
> for tk in d: print(tk, tk.head, tk.head.i)

is always

> The The 0
> cat cat 1

or a value error.

I'm using spacy 3.0.6.

I'm completely stumped and it would be great if the docs could provide more details of how to do this correctly.

Thank you!

## Which page or section is this issue related to?
https://spacy.io/api/doc#init

**Comment by polm** [comment, marked as answer]
Sorry you're having trouble with this. 

So this should be issuing a warning or something, but if you pass heads without deps they are not used. This works the way you want it to:

```
import spacy
from spacy.tokens import Doc

nlp = spacy.blank("en")

doc = Doc(nlp.vocab, ["The", "cat"], heads=[1, 1], deps=["xxx", "xxx"])

for word in doc:
    print(word, word.head)
```

**Reply by polm** [reply]
Oh yeah, we'd like to improve how this works in spaCy, but we're not sure about whether we should provide a default value for deps or give an error. To give us more perspective on the issue, can you clarify your use case? How do you end up with head information but not dep labels?

**Reply by pb-jeff-oneill** [reply]
Thank you!  I thought I had tried adding deps but clearly I didn't.

I actually have deps, but the reason I left them out is that this is for unit tests and I wanted to leave out details that don't matter for the tests.  I can add dummy values for the deps so that is an easy fix.

It would be helpful to add a note to the docs for the `heads` argument in `Doc.__init__` that `deps` are required.  :)
